### PR TITLE
Feature/i18n user prefs and state

### PR DESCRIPTION
**Pull request by skyeturriff**
### Intent

Adds i18n to user state and preferences ( for `user-prefs-schema.json` and and `user-state-schema.json`, plus a few additional string in some of the PreferencesPanes).

### Approach

Updated `generate-prefs.R` to emit Constants and Properties (i.e. `UserPrefsAccessorConstants.java`, `UserPrefsAccessorConstants_en.properties`, etc) files for all the user settings.

The `UserPrefsAccessor` and `UserStateAccessor` classes now pull i18n-ed strings from the emitted Constants Interfaces. All user-facing enumerations defned in the `json` files (should) have beeen handled as well.

### QA Notes

This can be tested by following the existing steps for updating user state and prefs. In  `generate-prefs.R`, you can also set the `prefixDefault` and `prefixProperties` to some easily recognizable value, to make changes stand out visually.

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

cc @gtritchie

### CODE REVIEW

File: src/cpp/session/resources/schema/user-prefs-schema.json [hunk 1]
```diff
@@ -641,6 +647,12 @@
             "title": "Source with echo by default",
             "description": "Whether to echo R code when sourcing it."
         },
+        "new_project_git_init": {
```
Comment (jmcphers) [inline src/cpp/session/resources/schema/user-prefs-schema.json:650]: This pref (and several others) were removed a while back; this change should not reintroduce them.

File: src/gwt/src/org/rstudio/studio/client/workbench/prefs/model/Prefs.java [hunk 2]
```diff
@@ -359,10 +359,12 @@ protected void doSetValue(JsObject root, String name, String value)
    public class EnumValue extends JsonValue<String>
    {
       private EnumValue(String name, String title, String description, 
-                        String[] values, String defaultValue)
+                        String[] values, String defaultValue, String[] readableValues)
       {
          super(name, title, description, defaultValue);
          allowedValues_ = values;
+         readableValues_ = readableValues;  // Human readable text describing the readable values
+         assert(allowedValues_.length == readableValues_.length);  // DEBUG: Is this a good idea to validate like this?  Probably need to do an explicit throw instead?
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/workbench/prefs/model/Prefs.java:367]: This is a private constructor so I'd suggest moving this check upstream, and adding a debug.warn statement that logs the enum in question to the console.

File: src/gwt/tools/generate-prefs.R [hunk 3]
```diff
@@ -177,30 +183,91 @@ generate <- function (schemaPath, className) {
...
+                                        camel, "Enum_", gsub("[^A-Za-z0-9_]", "_", enumVal), "=", prefixProperties, enumReadable
+                                     )
+                                  }, def[["enum"]], def[["enumReadable"]]), collapse = "\n"
+                                  ), "\n"
+         )
+      }
+
+      javaProperties <- paste0(javaProperties, "\n")
+
+      # DEBUG: Add entry here for the java constants and properties files.  Then change perfTitle to
```
Comment (jmcphers) [inline src/gwt/tools/generate-prefs.R:235]: Does this still need to be done?

File: src/gwt/tools/generate-prefs.R [hunk 4]
```diff
@@ -177,30 +183,91 @@ generate <- function (schemaPath, className) {
...
+
+      javaConstants <- paste0(javaConstants, "\n")
+
+      # Add title and description entries for the java properties file
+      javaProperties <- paste0(javaProperties,
+         commentProperties,
+         camel, "Title = ", prefixProperties, prefTitle, "\n",
+         camel, "Description = ", prefixProperties, def[["description"]], "\n")
+
+      if (!is.null(def[["enumReadable"]]))
```
Comment (jmcphers) [inline src/gwt/tools/generate-prefs.R:221]: I think this would be a good place to validate that enumReadble has the same length as the enum.

**Review by jmcphers — CHANGES_REQUESTED** [review]

File: src/gwt/tools/generate-prefs.R (hunk 3 above)
Comment (skyeturriff) [reply src/gwt/tools/generate-prefs.R:235]: I believe this is a left over comment from Andrew's work on this. All files are being generated, so I'll remove it

File: src/gwt/tools/generate-prefs.R (hunk 4 above)
Comment (skyeturriff) [reply src/gwt/tools/generate-prefs.R:221]: I've added a check near the beginning of the loop, to prevent unnecessary work from being done. This is under the assumption that if a pref has an invalid attribute, a msg should be logged, the generation of that pref should be skipped, and the script should continue with the next pref. Let me know if some other behaviour is desired! (i.e. halting the whole script)

File: src/cpp/session/resources/schema/user-prefs-schema.json (hunk 1 above)
Comment (skyeturriff) [reply src/cpp/session/resources/schema/user-prefs-schema.json]: Removed this one, and found two others (also removed)

File: src/cpp/session/resources/schema/user-prefs-schema.json [hunk 5]
```diff
@@ -230,9 +230,9 @@
         },
         "auto_detect_indentation": {
             "type": "boolean",
-            "default": false,
-            "title": "Auto-detect indentation in files",
-            "description": "Whether to automatically detect indentation settings from file contents."
+            "default": true,
+            "title": "Autodetect indentation in files",
+            "description": "When enabled, the indentation for documents not part of an RStudio project will be automatically detected."
```
Comment (gtritchie) [inline src/cpp/session/resources/schema/user-prefs-schema.json:235]: Why are these three lines changed? These don't match what is currently in `main` (maybe this got synced with an older state, or branch, or something?)

File: src/cpp/session/resources/schema/user-prefs-schema.json [hunk 6]
```diff
@@ -268,16 +268,18 @@
             "type": "boolean",
             "default": false,
             "title": "Continue comments after adding new line",
-            "description": "Whether continue comments (by inserting the comment character) after adding a new line."
+            "description": "Whether to continue comments (by inserting the comment character) after adding a new line.  Press Shift + Enter to exit a comment."
```
Comment (gtritchie) [inline src/cpp/session/resources/schema/user-prefs-schema.json:271]: Not sure why this is changed. Adding word "to" makes sense, but not sure why "Press Shift + Enter..." was added.

File: src/cpp/session/resources/schema/user-prefs-schema.json [hunk 7]
```diff
@@ -268,16 +268,18 @@
...
         },
         "highlight_web_link": {
             "type": "boolean",
             "default": true,
-            "description": "Whether web links in comments are clickable."
+            "title": "Hyperlinks in comments will be underlined and clickable",
+            "description": "Hyperlinks in comments will be underlined and clickable"
```
Comment (gtritchie) [inline src/cpp/session/resources/schema/user-prefs-schema.json:277]: Are these changes necessary or maybe part of a bad merge?

File: src/cpp/session/resources/schema/user-prefs-schema.json [hunk 8]
```diff
@@ -565,7 +571,7 @@
         },
         "limit_visible_console": {
             "type": "boolean",
-            "default": false,
+            "default": true,
```
Comment (gtritchie) [inline src/cpp/session/resources/schema/user-prefs-schema.json:574]: Don't change this.

File: src/cpp/session/resources/schema/user-prefs-schema.json [hunk 9]
```diff
@@ -321,8 +323,9 @@
         "surround_selection": {
             "type": "string",
             "enum": ["never", "quotes", "quotes_and_brackets"],
+            "enumReadable": ["Never", "Quotes", "Quotes & Brackets"],
             "default": "quotes_and_brackets",
-            "title": "Surround selections with",
+            "title": "Surround selections with delimiter insertion",
```
Comment (gtritchie) [inline src/cpp/session/resources/schema/user-prefs-schema.json:328]: Though this seems like a reasonable correction, let's not mess with this unless absolutely necessary for this PR.

**Review by gtritchie — CHANGES_REQUESTED** [review]

**Comment by skyeturriff** [comment]
Sorry @gtritchie ignore that review request, clicked by mistake

**Comment by gtritchie** [comment]
> Sorry @gtritchie ignore that review request, clicked by mistake

I JUST sent an email asking about that. :-)

**Comment by skyeturriff** [comment]
Okay @gtritchie, NOW it's ready 😅

**Review by gtritchie — APPROVED** [review]
